src/utils/call_plots.jl

```diff
@@ -0,0 +1,180 @@
+    Colors.RGBA(1, 1, 0.5, 0.6), # light yellow
+    Colors.RGBA(0.27, 0.5, 0.7, 0.9), # steel blue
+    Colors.RGBA(1, 0.757, 0.15, 01), # canary yellow
+    Colors.RGBA(0.8, 0.6, 0.3, 1), :red] # khaki
+    
+    fuels = ["Nuclear", "Coal", "Hydro", "Gas_CC",
+             "Gas_CT", "Storage", "Oil_ST", "Oil_CT",
+             "Sync_Cond", "Wind", "Solar", "CSP", "curtailment"]
+  
+    color_fuel = DataFrames.DataFrame(fuels = fuels, colors = color_range)
+  
+    stack = get_stacked_aggregation_data(res, generator_dict)
+    bar = get_bar_aggregation_data(res, generator_dict)
+    default =  [(color_fuel[findall(in(["$(bar.labels[1])"]),
+    color_fuel.fuels), :][:,:colors])[1]]
+    for i in 2:length(bar.labels)
+      specific_color = (color_fuel[findall(in(["$(bar.labels[i])"]),
+                        color_fuel.fuels), :][:,:colors])[1]
+      default = [default specific_color]
+      
+    end
+  
+    seriescolor = (get(kwargs, :seriescolor, default))
+    P1 = RecipesBase.plot(stack; seriescolor = seriescolor)
+    P2 = RecipesBase.plot(bar; seriescolor = seriescolor)
+ 
+    display(P1)
+    display(P2)
+end
+
+function fuel_plot(res::PSI.SimulationResults, generator_dict::Dict; kwargs...)
+    results = OperationsProblemResults(res.variables, res.total_cost, 
+    res.optimizer_log, res.time_stamp)
+    fuel_plot(results, generator_dict; kwargs...)
+end
+
+function fuel_plot(res::PSI.DualResults, generator_dict::Dict; kwargs...)
+    results = OperationsProblemResults(res.variables, res.total_cost, 
+    res.optimizer_log, res.time_stamp)
+    fuel_plot(results, generator_dict; kwargs...)
+end
+=#
+"""
+   bar_plot(OperationsProblemResults)
+  
+This function plots a bar plot for the generators in each variable within
+the results variables dictionary, and makes a bar plot for all of the variables.
+
+# Arguments
+- `res::OperationsProblemResults= results`: results to be plotted
+
+# Example
+
+```julia
+results = solve_op_problem!(OpProblem)
+bar_plot(results)
+```
+
+# Accepted Key Words
+plot attributes, such as seriescolor = [:red :blue :orange]
+will override the default series color
+"""
+
+function bar_plot(res::OperationsProblemResults; kwargs...)
+    default = hcat([Colors.RGBA(0.7,0.1,0.1,0.95)], # maroon
+              [Colors.RGBA(0,0,0,0.8)], [:lightblue], # Dark gray
+              [Colors.RGBA(0.33,0.42,0.18,0.9)], [:pink], # olive green
+              [Colors.RGBA(0.93,0.46,0,1)], [Colors.RGBA(0.56,0.28,0.54,1)], # orange, orchid
+              [Colors.RGBA(0.9,0.5,0.6,0.80)],  # dark pink
+              [Colors.RGBA(1, 1, 0.5, 0.6)], # light yellow
+              [Colors.RGBA(0.27, 0.5, 0.7, 0.9)], # steel blue
+              [Colors.RGBA(1, 0.757, 0.15, 01)], # canary yellow
+              [Colors.RGBA(0.8, 0.6, 0.3, 1)], [:red]) # khaki
+    seriescolor = get(kwargs, :seriescolor, default)
+    key_name = string.(collect(keys(res.variables)))
+    for name in key_name
```

Comment:
Performance doesn't matter here, but it would be good to generally not create temporary arrays when you don't need them.  You could just iterate over the call to keys().